### PR TITLE
Add chart for Atlantis:

### PR DESCRIPTION
**Pull request by jkodroff**
https://runatlantis.io
Signed-off-by: Josh Kodroff <jkodroff@2ndwatch.com>

#### What this PR does / why we need it:

There's no Helm chart for Atlantis, and it's a useful tool.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by sc250024** [comment]
Yes. So much yes.

File: incubator/atlantis/.helmignore
Comment (davidkarlsen) [inline incubator/atlantis/.helmignore]: aim straight for the stable repo

File: incubator/atlantis/templates/statefulset.yaml
Comment (davidkarlsen) [inline incubator/atlantis/templates/statefulset.yaml]: go for v1

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: incubator/atlantis/.helmignore
Comment (lkysow) [reply incubator/atlantis/.helmignore]: It has been in production for >2 years... :)

**Comment by jkodroff** [comment]
@davidkarlsen One question: Could you verify the minimum version of Kubernetes?  It's a total shot in the dark - I have no idea what the actual required version is because I don't know when StatefulSets were introduced.

Thanks so much for the timely review!

File: incubator/atlantis/.helmignore
Comment (jkodroff) [reply incubator/atlantis/.helmignore]: Done.

File: incubator/atlantis/templates/statefulset.yaml
Comment (jkodroff) [reply incubator/atlantis/templates/statefulset.yaml]: Done.

**Comment by mootpt** [comment]
@jkodroff where is the secret managed for aws? I didn't see it anywhere.

**Comment by mootpt** [comment]
@jkodroff think you are missing secrets for gitconfig and aws. Something like:

```
apiVersion: v1
kind: Secret
metadata:
  name: {{ template "atlantis.fullname" . }}-aws
  labels:
    app: {{ template "atlantis.name" . }}
    chart: {{ template "atlantis.chart" . }}
    release: {{ .Release.Name }}
    heritage: {{ .Release.Service }}
data:
    {{- if .Values.aws.config }}
    config: {{ .Values.aws.config | b64enc }}
    {{- end}}
    {{- if .Values.aws.credentials }}
    credentials: {{ .Values.aws.credentials | b64enc }}
    {{- end}}
```

**Comment by sstarcher** [comment]
For users who do not use ingress it's helpful to include annotations on the service.
https://github.com/helm/charts/blob/master/stable/grafana/templates/service.yaml#L13

By that I mean I need that

**Comment by sstarcher** [comment]
@jkodroff Kubernetes 1.9 statefulset was GA and Kubernetes 1.5 it was beta.

You are specifically calling out the GA release of stateful sets so your minimum version should be 1.9
```
apiVersion: apps/v1
```

**Comment by zytek** [comment]
Are FS permissions legit?
```
[ERROR] initializing server: initializing terraform: unable to create terraform 
plugin cache directory at "plugin-cache": mkdir /atlantis-data/plugin-cache: permission denied
```

**Comment by jkodroff** [comment]
Thanks for the useful comments and my apologies for the neglect on this PR - I got pulled away on some client work.

I'm at Hashiconf for most of next week, but wrapping up this PR is my top priority as soon as I have time.  Thank you for your patience.

**Comment by jkodroff** [comment]
@mootpt Done.
@sstarcher Done.
@zytek Done.
@davidkarlsen Done.

@davidkarlsen If this all looks good, should I open a new PR against a single, signed commit?

**Comment by sstarcher** [comment]
@jkodroff no need to open a new PR you can squash your changes and sign the commit and force push them up.

File: stable/atlantis/templates/statefulset.yaml
Comment (sstarcher) [inline stable/atlantis/templates/statefulset.yaml:4]: Also add the annotation option to the stateful set.  That will allow people to use kube2iam for AWS credentials.

File: stable/atlantis/templates/statefulset.yaml
Comment (sstarcher) [inline stable/atlantis/templates/statefulset.yaml]: I believe these should be fullname.  If you were to install this into the same namespace twice this would collide and error

File: stable/atlantis/templates/statefulset.yaml
Comment (sstarcher) [inline stable/atlantis/templates/statefulset.yaml]: fullname

File: stable/atlantis/templates/secret-aws.yaml
Comment (sstarcher) [inline stable/atlantis/templates/secret-aws.yaml]: all metadata.names should be fullname

**Review by sstarcher — CHANGES_REQUESTED** [review]

**Comment by sstarcher** [comment]
Additional need will be for SSL configuration, but that can be handled on a new PR.

File: stable/atlantis/templates/service.yaml
Comment (sstarcher) [inline stable/atlantis/templates/service.yaml]: this needs to have no indentation and should be 6

**Review by sstarcher — CHANGES_REQUESTED** [review]

**Comment by sstarcher** [comment]
I'll need this as part of the service or I can do a separate PR 

https://github.com/helm/charts/blob/cbd5e811a44c7bac6226b019f1d1810ef5ee45fa/stable/cluster-autoscaler/templates/service.yaml#L20

**Comment by sstarcher** [comment]
sorry ment - https://github.com/helm/charts/blob/cbd5e811a44c7bac6226b019f1d1810ef5ee45fa/stable/cluster-autoscaler/templates/service.yaml#L23

File: stable/atlantis/templates/statefulset.yaml
Comment (jkodroff) [reply stable/atlantis/templates/statefulset.yaml]: Correct, and done.

**Comment by sstarcher** [comment]
It would be useful to apply a checksum so the pod gets restarted on configuration updates
https://github.com/helm/charts/blob/9f78ebf325952ffc2da03cf863c6da2fd871b366/stable/kuberos/templates/deployment.yaml#L10

File: stable/atlantis/templates/service.yaml
Comment (sstarcher) [reply stable/atlantis/templates/service.yaml]: sorry 4 is fine, but the toYaml block needs to be left aligned.  If you have more then one annotation it would currently error.

File: stable/atlantis/values.yaml
Comment (sstarcher) [inline stable/atlantis/values.yaml]: Any reason to have such a long initial delay?  Atlantis seems to startup fast.

**Review by sstarcher — CHANGES_REQUESTED** [review]

**Comment by sstarcher** [comment]
@jkodroff I went ahead and made some of the changes approve I requested.  If you don't mind pulling them into your PR that would be great if not I'll put in another PR after you merge this one.

This is based off your latest commit - https://github.com/sstarcher/charts/commit/5d8902b42a1b9f14d445895de7e9a8a1390412c2

**Comment by mikeraimondi** [comment]
Excited to see this merged!

**Comment by halcyondude** [comment]
Yes please, we have use for this :)

File: stable/atlantis/values.yaml
Comment (lkysow) [reply stable/atlantis/values.yaml]: Agreed, the default is fine since Atlantis starts in <1s. The periodSeconds is set to 60 just because Atlantis isn't high throughput so there's no point hammering it with health checks

File: stable/atlantis/templates/statefulset.yaml
Comment (MattiasGees) [inline stable/atlantis/templates/statefulset.yaml:192]: Can you also add storageClass like https://github.com/helm/charts/blob/9897667e80066a3ec0e4a0c98d2c45400784e9d9/stable/consul/templates/consul-statefulset.yaml#L190-L193 ?

Makes using another class easier.

File: stable/atlantis/templates/statefulset.yaml
Comment (MattiasGees) [inline stable/atlantis/templates/statefulset.yaml]: Maybe make this configurable?

File: stable/atlantis/templates/statefulset.yaml
Comment (lkysow) [inline stable/atlantis/templates/statefulset.yaml]: this needs to be `atlantis.fullname`

File: stable/atlantis/templates/statefulset.yaml
Comment (lkysow) [inline stable/atlantis/templates/statefulset.yaml:102]: `atlantis.fullname`

File: stable/atlantis/templates/statefulset.yaml
Comment (lkysow) [inline stable/atlantis/templates/statefulset.yaml]: `atlantis.fullname`

File: stable/atlantis/templates/statefulset.yaml
Comment (lkysow) [inline stable/atlantis/templates/statefulset.yaml]: `atlantis.fullname`

File: stable/atlantis/templates/statefulset.yaml
Comment (lkysow) [inline stable/atlantis/templates/statefulset.yaml]: `atlantis.fullname`

**Review by lkysow — CHANGES_REQUESTED** [review]

**Comment by jkodroff** [comment]
> @jkodroff I went ahead and made some of the changes approve I requested. If you don't mind pulling them into your PR that would be great if not I'll put in another PR after you merge this one.
> 
> This is based off your latest commit - [sstarcher@5d8902b](https://github.com/sstarcher/charts/commit/5d8902b42a1b9f14d445895de7e9a8a1390412c2)

@sstarcher Let's do these as an additional PR.  They look good, though.

File: stable/atlantis/values.yaml
Comment (jkodroff) [reply stable/atlantis/values.yaml]: Changed to 5s.

**Comment by jkodroff** [comment]
@sstarcher Would you like to be included on the list of reviewers and maintainers?  You seem to know what you're doing more than I.

**Comment by sstarcher** [comment]
@jkodroff sure, and you are doing a great job.

**Review by lkysow — APPROVED** [review]
Thanks for making the changes.

File: stable/atlantis/templates/service.yaml
Comment (jkodroff) [reply stable/atlantis/templates/service.yaml]: Fixed.  It's actually `annotations` that was at the wrong indentation level.

File: stable/atlantis/templates/statefulset.yaml
Comment (jkodroff) [reply stable/atlantis/templates/statefulset.yaml]: Done

**Comment by jkodroff** [comment]
@davidkarlsen Can we get an LGTM?  Would love to get this merged this week if possible.  (With several improvements soon to follow.)

File: stable/atlantis/templates/statefulset.yaml
Comment (jkodroff) [reply stable/atlantis/templates/statefulset.yaml:4]: Done.

**Comment by zytek** [comment]
Two issues still present:
- default values lack .Values.statefulset.annotations
- those annotations are on Statefulset object ,but are not attached to Pod template so I cannot assign kube2iam roles to atlantis

File: stable/atlantis/Chart.yaml
Comment (davidkarlsen) [inline stable/atlantis/Chart.yaml]: target 1.0.0

File: stable/atlantis/README.md
Comment (davidkarlsen) [inline stable/atlantis/README.md]: helm install requires a name of the chart too

File: stable/atlantis/templates/_helpers.tpl
Comment (davidkarlsen) [inline stable/atlantis/templates/_helpers.tpl:26]: also add https://github.com/helm/helm/blob/master/docs/chart_best_practices/rbac.md and apply that serviceaccount for the deployment/statefulset.

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

File: stable/atlantis/Chart.yaml
Comment (lkysow) [inline stable/atlantis/Chart.yaml]: There's been a new release so this should be `v0.4.11`

File: stable/atlantis/values.yaml
Comment (lkysow) [inline stable/atlantis/values.yaml]: v0.4.11 now

**Comment by antonosmond** [comment]
Hey I had atlantis working in kubernetes using the statefulset.yaml [here](https://www.runatlantis.io/docs/deployment.html#kubernetes)  plus the relevant changes to add AWS & Google creds and then I switched to the helm chart from [here](https://github.com/runatlantis/atlantis/tree/master/helm/atlantis) using the latest v0.4.11. It's not picking up the google creds. They are correctly mounted at `/etc/credentials/service-account.json` but I can't see anything setting an env var e.g: `GOOGLE_APPLICATION_CREDENTIALS` to that path. Am I missing something or this something that's been missed in the helm chart?

File: stable/atlantis/values.yaml
Comment (jkodroff) [reply stable/atlantis/values.yaml]: Done.

File: stable/atlantis/Chart.yaml
Comment (jkodroff) [reply stable/atlantis/Chart.yaml]: Done.

File: stable/atlantis/Chart.yaml
Comment (jkodroff) [reply stable/atlantis/Chart.yaml]: Done.

File: stable/atlantis/README.md
Comment (jkodroff) [reply stable/atlantis/README.md]: If we're testing the chart, I assume that it's going to be called from the repo, in the same directory as the chart.  (That's why I have `. --debug` instead of `stable/atlantis --debug`.)

Please let me know if `.` is ok, and if not, I'll change it to `stable/atlantis`.

**Comment by jkodroff** [comment]
@zytek I believe your concerns should be addressed now (from `values.yaml`):

```yaml
podTemplate:
  annotations: {}
    # kube2iam example:
    # iam.amazonaws.com/role: role-arn

statefulSet:
  annotations: {}
```

File: stable/atlantis/templates/_helpers.tpl
Comment (jkodroff) [reply stable/atlantis/templates/_helpers.tpl:26]: @lkysow @davidkarlsen I'm fairly new to (i.e. ignorant of) RBAC resources, so I'd appreciate a sanity check of my understanding:

As far as I can tell, the only thing that would need to (optionally) be created is a `ServiceAccount` with no permissions because Atlantis does not perform any actions against the k8s API.

This is what I did in the latest push.

**Comment by jkodroff** [comment]
@antonosmond My apologies for that.  Could you paste the YAML snippet that fixed the issue for you below?  I'm not familiar with Google Cloud setup for Atlantis and I don't see anything relating to Google in the Atlantis docs, so I'm not exactly sure what the desired end state of the manifest should be.

**Comment by jkodroff** [comment]
@davidkarlsen Ball is back in your court as I believe I have addressed all outstanding issues.

@antonosmond If this PR gets merged before you reply, I'll be sure to add those changes as soon as I hear from you.

**Comment by antonosmond** [comment]
@jkodroff after having another look through the chart it appears it deals with multiple Google service accounts so setting the env var wouldn't be able to handle that. I guess we can just deal with setting the service account path by adding it explicitly in the Google terraform provider so maybe it's best to ignore my previous comments for now.

File: stable/atlantis/README.md
Comment (davidkarlsen) [reply stable/atlantis/README.md]: stable/atlantis is what it will become when released

File: stable/atlantis/templates/_helpers.tpl
Comment (davidkarlsen) [reply stable/atlantis/templates/_helpers.tpl:26]: correct

**Comment by sstarcher** [comment]
@jkodroff looks like it's failing lint atm.

File: stable/atlantis/templates/ingress.yaml
Comment (lkysow) [inline stable/atlantis/templates/ingress.yaml:27]: Should match the name of the service `{{ template "atlantis.fullname" . }}`

File: stable/atlantis/templates/ingress.yaml
Comment (lkysow) [inline stable/atlantis/templates/ingress.yaml:28]: Should match the port of the service: `{{ .Values.service.port }}`

**Comment by jkodroff** [comment]
@sstarcher I don't see it failing lint.  Is there something I'm missing?

```bash
➜  atlantis git:(incubator-atlantis) pwd
/Users/jkodroff/src/charts/stable/atlantis
➜  atlantis git:(incubator-atlantis) helm lint
==> Linting .
Lint OK

1 chart(s) linted, no failures
➜  atlantis git:(incubator-atlantis) helm version
Client: &version.Version{SemVer:"v2.12.0", GitCommit:"d325d2a9c179b33af1a024cdb5a4472b6288016a", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
```

File: stable/atlantis/templates/ingress.yaml
Comment (jkodroff) [reply stable/atlantis/templates/ingress.yaml:27]: I believe it already does, per line 2 of this same file.  Please mark resolved or correct me if I've got it wrong.

**Comment by sstarcher** [comment]
@jkodroff look at the github status checks here the lint done by helm is not the same as the tests.

https://circleci.com/gh/helm/charts/40919?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by sstarcher** [comment]
```
Linting 'stable/atlantis/values.yaml'...
stable/atlantis/values.yaml
  154:8     error    no new line character at the end of file  (new-line-at-end-of-file)
```

**Comment by jkodroff** [comment]
@sstarcher Linting fixed.

File: stable/atlantis/README.md
Comment (davidkarlsen) [reply stable/atlantis/README.md]: still "." ?

Comment (jkodroff) [reply stable/atlantis/README.md]: Fixed.

File: stable/atlantis/templates/_helpers.tpl
Comment (davidkarlsen) [reply stable/atlantis/templates/_helpers.tpl:26]: can you add documentation for the new fields in the readme too?

Comment (jkodroff) [reply stable/atlantis/templates/_helpers.tpl:26]: Done (and alphabetized for less git merge headaches in the future).

**Review by davidkarlsen — APPROVED** [review]

**Comment by cpanato** [comment]
looks like some flags need to be set 
```
--gh-user/--gh-token or --gitlab-user/--gitlab-token or --bitbucket-user/--bitbucket-token must be set
```
can you add a ci folder with the values that need to be set?

**Comment by jkodroff** [comment]
@davidkarlsen Looks like we just need a couple of labels and we're good to go.

**Pull request merged**
